### ISSUE TITLE
Not showing up in node_modules/.bin until npm rebuild

### ISSUE THREAD

**Issue by mtfurlan** [sanack/node-jq#669]
## Description
When I install `node-jq`, the symlink doesn't show up in `node_modules/.bin` until I do `node-rebuild`, so `npx node-jq whatever` fails.

## Reproduction
create new directory
```
$ npm install node-jq tsc
...
$ ls node_modules/.bin/
ndh@  seek-bunzip@  seek-table@  tsc@  uuid@  which@
$ npm rebuild
rebuilt dependencies successfully
$ ls node_modules/.bin/
ndh@  node-jq@  seek-bunzip@  seek-table@  tsc@  uuid@  which@
```
Notice how `tsc` showed up the first time, `node-jq` only after `npm rebuild`.

If I run with `--loglevel verbose` both the install and the rebuild have
```
npm info run node-jq@4.3.0 postinstall node_modules/node-jq npm run install-binary
npm info run node-jq@4.3.0 postinstall { code: 0, signal: null }
```

## Environment
All testing was done with node-jq 4.3.0

Systems tested:
* Debian 12 bookworm linux 6.1.0-15-amd64
  * node/npm versions:
    * v21.6.2 / 10.5.0
    * v20.11.1 / 10.2.4
    * v18.19.1 / 10.2.4
    * v16.20.2 / 8.19.4
* macos sonoma 14.3.1 darwin 23.3.0
 * node v20.5.1
 * npm 10.5.0
 
 
## The bit where it gets weird
If I modify the `node-jq` `package.json` bin field from
```diff
diff --git a/package.json b/package.json
index 94a445c..e07284e 100644
--- a/package.json
+++ b/package.json
@@ -8,7 +8,8 @@
     "url": "https://github.com/sanack/node-jq"
   },
   "bin": {
-    "node-jq": "bin/jq"
+    "node-jq": "bin/jq",
+    "foobar": "bin/jq"
   },
   "scripts": {
     "pretest": "npm run install-binary",
```
it works as expected.

So I tried making a test module with a single thing in the bin field, but it worked as expected.
```
{
  "name": "testpackage",
  "version": "1.0.0",
  "description": "",
  "bin": {
    "bar": "foo"
  }
}
```
 

---
So, any idea what might be happening?
I can do bug report against npm if we can figure out how to reproduce.

**Comment by davesnx** [comment]
Whoa, this looks bad and the only thing that might be happening is the path? Checking tsc they use `"./bin/tsc"` https://github.com/microsoft/TypeScript/blob/main/package.json#L25 and node-jq uses `"bin/jq"`.

**Comment by mtfurlan** [comment]
I tried that earlier, and it didn't work.

Though now whenever I install the node-jq package from a local directory it does the symlink, while the npm package still isn't.
But it is a different process because from a local directory install also installs dev dependancies I think?

So I dunno what changed.
I'll poke more monday.

Can you replicate any of this, or is it only being weird for me and a coworker?

**Comment by davesnx** [comment]
That's interesting, it should work indeed. Want to push a PR so we can play with a prerelease version?

**Comment by davesnx** [comment]
You can test the new version with the fix node-jq@4.3.1, will keep it open (was automatically closed)

**Comment by mtfurlan** [comment]
It works!
